server/routes/publicLists.js

```diff
@@ -0,0 +1,51 @@
+const express = require("express");
+const router = express.Router();
+const UserListModel = require("../models/userListModel");
+const db = require("../models");
+
+const getLists = async (req, res) => {
+  const userId = req.params.userId;
+  try {
+    const results = await UserListModel.findAll({
+      attributes: ["id", "title", "items", "imageUrl", "isPrivate", "userId"],
+      where: { userId: userId, isPrivate: false },
+      order: ["id"],
+    });
+    res.status(200).send(results);
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const getProductsOfList = async (req, res) => {
+  const userId = req.params.userId;
+  const listId = req.params.listId;
+  try {
+    const listFound = await UserListModel.findOne({
```

Comment:
See if you can make this a transaction. My lists route has a couple good examples of transactions that you can use. I'm not sure if committing or rolling back will be necessary since these are just selections. I can help look into it if it becomes an issue.